pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,252 @@
+		"--keyring=/etc/ceph/keyring-store/keyring",
+		"--log-to-stderr=true",
+		"--err-to-stderr=true",
+		"--mon-cluster-log-to-stderr=true",
+		"--log-stderr-prefix=debug",
+		"--default-log-to-file=false",
+		"--default-mon-cluster-log-to-file=false",
+		"--mon-host=$(ROOK_CEPH_MON_HOST)",
+		"--mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS)",
+		fmt.Sprintf("--id=%s", goodMon),
+		"--foreground",
+		"--public-addr=%s", goodMonPublicIp,
+		fmt.Sprintf("--setuser-match-path=/var/lib/ceph/mon/ceph-%s/store.db", goodMon),
+		"--public-bind-addr=$(ROOK_POD_IP)",
+		"--extract-monmap=%s", monmapPath,
+	}
+	exec.RunCommandInMonPod(context, fmt.Sprintf("rook-ceph-mon-%s_mon-debug", goodMon), "ceph-mon", args, clusterNamespace)
+
+	fmt.Println("Printing monmap")
+	exec.RunCommandInMonPod(context, fmt.Sprintf("rook-ceph-mon-%s_mon-debug", goodMon), "monmaptool", []string{fmt.Sprintf("--print %s", monmapPath)}, clusterNamespace)
+
+	// remove all the mons except the good one
+	for _, badMonId := range badMons {
+		fmt.Printf("Removing mon %s.\n", badMonId)
+		exec.RunCommandInMonPod(context, fmt.Sprintf("rook-ceph-mon-%s_mon-debug", goodMon), "monmaptool", []string{fmt.Sprintf("%s", monmapPath), "--rm", badMonId}, clusterNamespace)
+	}
+
+	fmt.Println("Injecting the monmap")
+	args = []string{
+		fmt.Sprintf("--fsid=%s", cephFsid),
+		"--keyring=/etc/ceph/keyring-store/keyring",
+		"--log-to-stderr=true",
+		"--err-to-stderr=true",
+		"--mon-cluster-log-to-stderr=true",
+		"--log-stderr-prefix=debug",
+		"--default-log-to-file=false",
+		"--default-mon-cluster-log-to-file=false",
+		"--mon-host=$(ROOK_CEPH_MON_HOST)",
+		"--mon-initial-members=$(ROOK_CEPH_MON_INITIAL_MEMBERS)",
+		fmt.Sprintf("--id=%s", goodMon),
+		"--foreground",
+		"--public-addr=%s", goodMonPublicIp,
+		fmt.Sprintf("--setuser-match-path=/var/lib/ceph/mon/ceph-%s/store.db", goodMon),
+		"--public-bind-addr=$(ROOK_POD_IP)",
+		"--inject-monmap=%s", monmapPath,
+	}
+	exec.RunCommandInMonPod(context, fmt.Sprintf("rook-ceph-mon-%s_mon-debug", goodMon), "ceph-mon", args, clusterNamespace)
+
+	fmt.Println("Finished updating the monmap!")
+
+	fmt.Println("Printing monmap")
+	exec.RunCommandInMonPod(context, fmt.Sprintf("rook-ceph-mon-%s_mon-debug", goodMon), "monmaptool", []string{fmt.Sprintf("--print %s", monmapPath)}, clusterNamespace)
+
+	fmt.Println("Restoring the mons in the rook-ceph-mon-endpoints configmap to the good mon")
+	monCm.Data[goodMon] = fmt.Sprintf("%s:%s", goodMonPublicIp, goodMonPort)
+
+	monCm, err = context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Update(ctx.TODO(), monCm, v1.UpdateOptions{})
+	if err != nil {
+		log.Fatalf("failed to update mon configmap %s %v", MonConfigMap, err)
+	}
+
+	fmt.Printf("Stopping the debug pod for mon %s.\n", goodMon)
+	debug.StopDebug(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", goodMon))
+
+	fmt.Println("Check that the restored mon is responding")
+	maxRetries := 1
+
+	for maxRetries == 20 {
```

Comment:
Wherever we have a for loop, likely should be factored out to a helper. This might also make sense as a helper in the exec package to wait for a ceph command to succeed.

Reply:
done